File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.Xpo
Imports System.Data.SqlClient
Imports DevExpress.Data.Filtering

Namespace CustomCommand
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitDAL()

			InitializeComponent()
		End Sub

		Private Sub InitDAL()
			Dim connStr As String = "Data Source=(local);Initial Catalog=Northwind;Integrated Security=True;"
			Dim connection As SqlConnection = New SqlConnection(connStr)
			Dim provider As MyMsSqlConnectionProvider = New MyMsSqlConnectionProvider(connection, DevExpress.Xpo.DB.AutoCreateOption.SchemaAlreadyExists)
			XpoDefault.DataLayer = New SimpleDataLayer(provider)
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			xpCollection1.Criteria = New OperandProperty("OrderMonth") = New OperandValue(2)
		End Sub
	End Class
End Namespace

File: CS/CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Xpo;
using System.Data.SqlClient;
using DevExpress.Data.Filtering;

namespace CustomCommand {
    public partial class Form1 : Form {
        public Form1() {
            InitDAL();

            InitializeComponent();
        }

        private void InitDAL() {
            string connStr = "Data Source=(local);Initial Catalog=Northwind;Integrated Security=True;";
            SqlConnection connection = new SqlConnection(connStr);
            MyMsSqlConnectionProvider provider = new MyMsSqlConnectionProvider(connection, DevExpress.Xpo.DB.AutoCreateOption.SchemaAlreadyExists);
            XpoDefault.DataLayer = new SimpleDataLayer(provider);
        }

        private void Form1_Load(object sender, EventArgs e) {
            xpCollection1.Criteria = new OperandProperty("OrderMonth") == new OperandValue(2);
        }
    }
}